libraries/aws_s3_bucket_object.rb

```diff
@@ -0,0 +1,99 @@
+class AwsS3BucketObject < Inspec.resource(1)
+  name 'aws_s3_bucket_object'
+  desc 'Verifies settings for a s3 bucket object'
+  example "
+    describe aws_s3_bucket_object(name: 'bucket_name', key: 'file_name') do
+      it { should exist }
+      its('permissions.authUsers') { should be_in [] }
+      its('permissions.owner') { should be_in ['FULL_CONTROL'] }
+      its('permissions.everyone') { should be_in [] }
+    end
+  "
+
+  include AwsResourceMixin
+  attr_reader :name, :key, :id, :public, :permissions
+  alias public? public
+
+  def to_s
+    "S3 Bucket Object #{@key} (#{@name})"
+  end
+
+  private
+
+  def validate_params(raw_params)
+    validated_params = check_resource_param_names(
+      raw_params: raw_params,
+      allowed_params: [:name, :key, :id],
+      allowed_scalar_name: [:name, :key],
```

Comment:
Hrm, I can't imagine that worked.  

`allowed_scalar_name` and `allowed_scalar_type` let you nominate exactly one resource parameter to be used instead of the hash syntax.  Here, you're trying to use two.

You should just remove these.